server/src/main/java/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -1004,4 +1031,15 @@ public boolean hasPendingHaWork(long vmId) {
         List<HaWorkVO> haWorks = _haDao.listPendingHaWorkForVm(vmId);
         return haWorks.size() > 0;
     }
+
+    @Override
+    public boolean hasPendingMigrationsWork(long vmId) {
+        List<HaWorkVO> haWorks = _haDao.listPendingMigrationsForVm(vmId);
+        for (HaWorkVO work : haWorks) {
+            if (work.getTimesTried() < _maxRetries) {
```

Comment:
should this be `<=`, just check.

Reply:
We start at 0 and increment on each failure by 1. So if work.getTimesTried() == maxRetries then it should be ideally never be returned here as _haDao internally filters out all items in Done, Canceled, Error state. I've added a warning in else here so that we know about any possible cases which don't get updated to Done, Canceled, Error state.